examples/2d/2d_shapes.rs

```diff
@@ -12,42 +13,73 @@ fn main() {
         .run();
 }
 
-const X_EXTENT: f32 = 600.;
-
 fn setup(
     mut commands: Commands,
     mut meshes: ResMut<Assets<Mesh>>,
     mut materials: ResMut<Assets<ColorMaterial>>,
 ) {
     commands.spawn(Camera2dBundle::default());
 
+    struct Shape {
+        mesh: Mesh2dHandle,
+        transform: Transform,
+    }
```

Comment:
Having structs inside functions is a bit unconventional imo; I'd move this above `Setup`